### PR TITLE
[Tables] Removed TableEntity attribute wrapper

### PR DESCRIPTION
**Pull request by annatisch**
Fixes issue #18442

### CODE REVIEW

File: sdk/tables/azure-data-tables/azure/data/tables/_base_client.py [hunk 1]
```diff
@@ -372,6 +372,8 @@ def parse_connection_str(conn_str, credential, keyword_args):
             }
         except KeyError:
             credential = conn_settings.get("sharedaccesssignature")
+            # if "sharedaccesssignature" in conn_settings:
+            #     credential = AzureSasCredential(conn_settings['sharedaccesssignature'])
```
Comment (seankane-msft) [inline sdk/tables/azure-data-tables/azure/data/tables/_base_client.py:376]: Do we need this wrapper?

File: sdk/tables/azure-data-tables/tests/test_table.py [hunk 2]
```diff
@@ -473,8 +473,8 @@ def test_account_sas(self, tables_storage_account_name, tables_primary_storage_a
 
             # Assert
             assert len(entities) ==  2
-            assert entities[0].text ==  u'hello'
-            assert entities[1].text ==  u'hello'
+            assert entities[0]['text'] ==  u'hello'
+            assert entities[1]['text'] ==  u'hello'
```
Comment (seankane-msft) [inline sdk/tables/azure-data-tables/tests/test_table.py:477]: very nitty
```suggestion
            assert entities[0]['text'] == u'hello'
            assert entities[1]['text'] == u'hello'
```

**Review by seankane-msft — APPROVED** [review]
one question, one nit. LGTM

File: sdk/tables/azure-data-tables/azure/data/tables/_base_client.py (hunk 1 above)
Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/_base_client.py:376]: I'm somewhat undecided....
I've left it as a TODO for me to revisit before GA.

**Pull request merged**